### ISSUE TITLE
v1.3.0 contains incompatible libraries

### ISSUE THREAD

**Issue by irogers97** [DataDog/dd-opentracing-cpp#168]
root@ca09ece2fcaa:/# nginx -V
nginx version: nginx/1.18.0
built by gcc 8.3.0 (Debian 8.3.0-6)
built with OpenSSL 1.1.1d  10 Sep 2019
TLS SNI support enabled
configure arguments: --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --modules-path=/usr/lib/nginx/modules --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/cache/nginx/client_temp --http-proxy-temp-path=/var/cache/nginx/proxy_temp --http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp --http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp --http-scgi-temp-path=/var/cache/nginx/scgi_temp --user=nginx --group=nginx --with-compat --with-file-aio --with-threads --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_slice_module --with-http_ssl_module --with-http_stub_status_module --with-http_sub_module --with-http_v2_module --with-mail --with-mail_ssl_module --with-stream --with-stream_realip_module --with-stream_ssl_module --with-stream_ssl_preread_module --with-cc-opt='-g -O2 -fdebug-prefix-map=/data/builder/debuild/nginx-1.18.0/debian/debuild-base/nginx-1.18.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fPIC' --with-ld-opt='-Wl,-z,relro -Wl,-z,now -Wl,--as-needed -pie'

root@aa192c28036f:~# nginx
version mismatch: 2 != 3
2021/03/26 14:16:04 [error] 21#21: Failed to load tracing library /usr/local/lib/libdd_opentracing_plugin.so: opentracing: versions of opentracing libraries are incompatible
nginx: [error] Failed to load tracing library /usr/local/lib/libdd_opentracing_plugin.so: opentracing: versions of opentracing libraries are incompatible

**Comment by cgilmour** [comment]
Hi @irogers97, thanks for reporting this!

Which version of nginx-opentracing is being used?
This release updated to opentracing-cpp v1.6.0, so the nginx-opentracing module also needs to be built with that version of opentracing. Since 0.14.0, they provide two builds per nginx version. You'll need the ot16 build of it for nginx 1.18.0 [(link)](https://github.com/opentracing-contrib/nginx-opentracing/releases/download/v0.14.0/linux-amd64-nginx-1.18.0-ot16-ngx_http_module.so.tgz).

Can you try it and let us know if that helps?

**Comment by return-nil** [comment]
@cgilmour 

Thank you.
You are a genius
I got the same phenomenon in the following environment but solved it by using your comment ot16

```
bash-4.2# get_latest_release() { \
>         wget -qO- "https://api.github.com/repos/$1/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/'; \
>     }
bash-4.2# get_latest_release opentracing-contrib/nginx-opentracing
v0.14.0
bash-4.2# get_latest_release DataDog/dd-opentracing-cpp
v1.3.0
bash-4.2# nginx -v
nginx version: nginx/1.18.0
```

**Comment by Tyler-2** [comment]
Somewhere, perhaps in Datadog's own documentation, I got this blob. I have updated it to include the ot16 bits.

```
# Install nginx-opentracing
RUN get_latest_release() { \
  wget -qO- "https://api.github.com/repos/$1/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/'; \
  } && \
  NGINX_VERSION=`nginx -v 2>&1 > /dev/null | sed -E "s/^.*nginx\/(.*)/\\1/"`&& \
  OPENTRACING_NGINX_VERSION="$(get_latest_release opentracing-contrib/nginx-opentracing)" && \
  DD_OPENTRACING_CPP_VERSION="$(get_latest_release DataDog/dd-opentracing-cpp)" && \
  \
  wget https://github.com/opentracing-contrib/nginx-opentracing/releases/download/${OPENTRACING_NGINX_VERSION}/linux-amd64-nginx-${NGINX_VERSION}-ot16-ngx_http_module.so.tgz && \
  NGINX_MODULES=$(nginx -V 2>&1 | grep "configure arguments" | sed -n 's/.*--modules-path=\([^ ]*\).*/\1/p') && \
  tar zxvf linux-amd64-nginx-${NGINX_VERSION}-ot16-ngx_http_module.so.tgz -C "${NGINX_MODULES}" && \
  # Install Datadog module
  wget -O - https://github.com/DataDog/dd-opentracing-cpp/releases/download/${DD_OPENTRACING_CPP_VERSION}/linux-amd64-libdd_opentracing_plugin.so.gz | gunzip -c > /usr/local/lib/libdd_opentracing_plugin.so && \
  rm -rf /var/lib/apt/lists/*

```

**Comment by cgilmour** [comment]
Yup, there's a few places where `get_latest_release` is part of documentation or examples.
Although convenient, it's also fragile for the bits we don't control (eg: nginx-opentracing's naming conventions, version formats etc).
I'll fix it for now, but if a better solution comes along, will replace it.

**Comment by dgoffredo** [comment]
Caleb fixed this in #171. Thanks for reporting the issue.
